lossfunctions_ag.tex

```diff
@@ -0,0 +1,87 @@
+\chapter{Loss Functions}
+\label{ch:loss functions}\index{loss functions}
+
+The driving force of any mathematical optimization problem, especially in machine learning, is the loss function, measuring how well an algorithm is performing with regards to its target. In general terms, the evaluation for any learning problem is done by taking a predicted or estimated output and compare it to the relative real recorded outcome, typically in a dataset. The expected value of the loss is called the statistical risk. The formulation of the loss function is crucial and must be accurately defined according to the selected algorithm and the selected data for the particular optimization problem. The loss function must reproduce a numerical value that somehow reflects the level of achievement desired by the algorithm.  In machine learning we are typically attempting to find the parameters that fit our model and that minimizes the loss function. Minimizing the loss function is one of the fundamental foundations of statistical machine learning \citep{pmlr-v32-yanga14}. 
+
+\begin{marginfigure}%
+	\includegraphics[width=\linewidth]{loss_functions/cross_entropy.png}
+	\caption{Log Loss Function}
+	\label{fig:logloss}
+\end{marginfigure}
+
+
+The three canonical loss functions as described by Vapnik \citep{vapnik1998statistical} are:
+\begin{itemize} \item The 0-1 Loss – the simplest loss function mainly used in binary classification \begin{equation} L(y,\hat{y)} \end{equation}, resulting in an error of 0 if and only if $y == \hat{y}$. Otherwise the function will produce the value of 1. This does not however differentiate between different classes and types of errors. Variations of this function may thus be formulated by extending the simple loss function and make the resulting loss dependent on another function.
+\item The squared loss – the equation used for regression defined as \begin{equation} L_{sq}(y,\hat{y}) \end{equation}  where the resulting error value is given by \begin{equation} (y-\hat{y)}^{2} \end{equation} This is often the loss function of choice used in estimators such as linear regression and in many areas of machine learning \citep {Davidson-Pilon:2015:BMH:2851115}. One disadvantage of using the squared loss function is that it is sensitive to outliers, since the loss increases quadratically as the estimate moves away from the true value \citep {Davidson-Pilon:2015:BMH:2851115}. A loss function which is less sensitive to outliers is the absolute-loss function, given by: \begin{equation} L(y,\hat{y)}=|y-\hat{y|}\end{equation} In this case, only the absolute value is taken of the difference between the estimated and real value. However this equation is not smooth where $(y-\hat{y})$ and thus may prove to be difficult to differentiate. 
```

Comment:
`\hat{y|}` will typeset it over the y and the bar in correct.  `\hat{y}|` is what you want.  You have this in a large number of places.  fix them all.  Also, shows you haven't looked properly at the compiled document.

Reply:
ok replace term file